async-nats/tests/client_tests.rs

```diff
@@ -764,4 +769,62 @@ mod client {
         drop(servers.remove(0));
         rx.recv().await;
     }
+
+    #[tokio::test]
+    async fn multiple_auth_methods() {
+        use async_nats::ServerAddr;
+        let path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+
+        let mut servers = vec![
+            nats_server::run_basic_server(),
+            nats_server::run_server("tests/configs/jwt.conf"),
+            nats_server::run_server("tests/configs/token.conf"),
+        ];
+
+        // multiple auth methods
+        let client = async_nats::ConnectOptions::new()
+            .with_user_and_password("js".into(), "js".into())
+            .unwrap()
+            .with_token("s3cr3t".into())
+            .unwrap()
+            .with_credentials_file(path.join("tests/configs/TestUser.creds"))
+            .await
+            .unwrap()
```

Comment:
I'm not sure that's the best API, `with_credentials_file` is async where most of the other auth options are *not*. It would be best if we're consistent.